release_notes_generator/model/service_chapters.py

```diff
@@ -145,23 +163,50 @@ def __populate_pr(self, record: Record, nr: int) -> None:
         if record.is_merged_pr:
             # check record properties if it fits to a chapter: MERGED_PRS_WITHOUT_ISSUE
             if not record.pr_contains_issue_mentions and not record.contains_min_one_label(self.user_defined_labels):
+                if self._is_row_present(nr) and not self.duplicity_allowed():
+                    return
+
                 self.chapters[MERGED_PRS_WITHOUT_ISSUE_AND_USER_DEFINED_LABELS].add_row(nr, record.to_chapter_row())
+                self.used_record_numbers.append(nr)
 
             # check record properties if it fits to a chapter: MERGED_PRS_LINKED_TO_NOT_CLOSED_ISSUES
             if record.pr_contains_issue_mentions:
+                if self._is_row_present(nr) and not self.duplicity_allowed():
+                    return
+
                 self.chapters[MERGED_PRS_LINKED_TO_NOT_CLOSED_ISSUES].add_row(nr, record.to_chapter_row())
+                self.used_record_numbers.append(nr)
 
             if not record.is_present_in_chapters:
+                if self._is_row_present(nr) and not self.duplicity_allowed():
+                    return
+
                 self.chapters[OTHERS_NO_TOPIC].add_row(nr, record.to_chapter_row())
+                self.used_record_numbers.append(nr)
 
         # check record properties if it fits to a chapter: CLOSED_PRS_WITHOUT_ISSUE
         elif (
             record.is_closed
             and not record.pr_contains_issue_mentions
             and not record.contains_min_one_label(self.user_defined_labels)
         ):
+            if self._is_row_present(nr) and not self.duplicity_allowed():
+                return
+
             self.chapters[CLOSED_PRS_WITHOUT_ISSUE_AND_USER_DEFINED_LABELS].add_row(nr, record.to_chapter_row())
+            self.used_record_numbers.append(nr)
 
         else:
+            if self._is_row_present(nr) and not self.duplicity_allowed():
+                return
+
             # not record.is_present_in_chapters:
             self.chapters[OTHERS_NO_TOPIC].add_row(nr, record.to_chapter_row())
+            self.used_record_numbers.append(nr)
+
+    def _is_row_present(self, nr: int) -> bool:
```

Comment:
Just curious, is this a reason why is this method only protected and not private? Because in this class you have a private, protected and static public methods. But you do use all of them only in the class.

Reply:
I made the method private too. See commit bbae8c5ae5deaa5c98ba9a3efffdb20e5b0ebdc7.